engine/src/main/java/io/github/trimax/venta/engine/model/common/scene/Fog.java

```diff
@@ -0,0 +1,13 @@
+package io.github.trimax.venta.engine.model.common.scene;
+
+import io.github.trimax.venta.engine.enums.FogType;
+import org.joml.Vector3f;
+
+public record Fog(
+        Vector3f color,
+        float density
+) {
+    public static Fog exponential(final Vector3f color, final float density) {
+        return new Fog(color, density);
+    }
```

Comment:
This method is redundant. It reflects the constructor. We can just use `new Fog(color, density)` instead.